amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/activity/InterchangeUtils.java

```diff
@@ -834,4 +838,28 @@ private static Field getField(Class<?> clazz, String fieldname) {
         
         return null;
     }
+
+    public static ActivityInformation getActivityInformation(Long projectId) {
+        AmpActivityVersion project = loadActivity(projectId);
+        if (project == null) {
+            //so far project will never be null since an exception will be thrown
+            //I leave the code prepared to throw the appropriate response code
+            ApiErrorResponse.reportResourceNotFound(ActivityErrors.ACTIVITY_NOT_FOUND);
```

Comment:
`AuthRule.VIEW_ACTIVITY` will validate that the activity is correct, so if it was validated, then here would be an abnormal case, i.e. `RuntimeException`

Reply:
Please see comment above